### DISCUSSION TITLE
Wiki repo restructuring

### DISCUSSION THREAD

**Discussion by jlvandenhout** [iotaledger/iota-wiki#463] [Ideas]
In the same logic of having a separate organization for the Wiki, I'd like to propose a restructuring of repositories as well, to separate concerns and keep the Wiki and its content maintainable. I think a few things are important to do:

1. Require project documentation to be self contained. Currently some projects imported under the build section have dependencies outside their documentation directory.
2. Extract our theme components and style to a separate Wiki style repo and publish it as Docusaurus theme. This allows the Wiki build as well as other projects to use it.
3. Extract any graphical assets into a single logically structured repo and host them separately, this has the benefit that they will display properly however you view them (on the Wiki, GitHub editor, Wiki editor, any editor), artists to work in a single repo and we don't have to do copy magic using the cli.
4. Extract internal docs to a separate repo, and have i18n in that repo too.

This way we have single logical repos for:

- The main Wiki build system.
- Our theme overriding part of Docusaurus classic theme where needed.
- Docs including versions and translations where appropriate.
- Graphical assets.

**Comment by Dr-Electron** [comment]
So :wen: editor V2 which supports all of this? :trollface:

**Reply by jlvandenhout** [reply]
:wen: 48 hour days??

**Reply by Dr-Electron** [reply]
Soon. I talked with the Boss :trollface:

**Comment by Dr-Electron** [comment]
Can you elaborate on point 3? About which assets are we talking exactly?

**Reply by jlvandenhout** [reply]
I'd say everything that goes into the static directory currently. Maybe I oversimplified with a single repo, because we have mixed content from both the community and IF. We could create two repos, one community controlled and one IF controlled, with proper 
copyright licensing.

**Comment by jlvandenhout** [comment]
The main reason I like to separate all of this logically, is because artists and contributors have to go into this rather technical build system focused repo to change content or graphics. I'd rather have the build system, theme, textual content and graphical content separated into their own repos, so devs, contributers and artists have their own focused space to work in and track issues and PRs.
